src/main.rs

```diff
@@ -92,22 +90,8 @@ async fn main() -> anyhow::Result<()> {
 
     println!("sk8brd {}", env!("CARGO_PKG_VERSION"));
 
-    // Connect to the local SSH server
-    let tcp = TcpStream::connect(format!("{}:{}", args.farm, args.port))
-        .with_context(|| format!("Couldn't connect to {}:{}", args.farm, args.port))?;
-    let mut sess = Session::new()?;
-    sess.set_tcp_stream(tcp);
-    sess.handshake()?;
-
-    // Try to authenticate with the first identity in the agent.
-    sess.userauth_agent(USERNAME)
-        .with_context(|| format!("Couldn't authenticate as {USERNAME}"))?;
-
-    let mut chan = Arc::new(Mutex::new(sess.channel_session()?));
-    (*get_arc!(chan))
-        .exec(CDBA_SERVER_BIN_NAME)
-        .with_context(|| format!("Couldn't execute {CDBA_SERVER_BIN_NAME} on remote host"))?;
-
+    let mut sess = ssh_connect(args.farm, args.port).await?;
+    let mut chan = ssh_get_chan(&mut sess).await?;
```

Comment:
You are abstracting the ssh2 crate, but it's still all about the ssh, at least judging by the name. A proper abstraction should be a dynamic trait which can be implemented by SSH or by any other protocol handler.
CDBA-over-HTTPS. Sweet idea.

Reply:
The ssh part is abstracted such that anyone that wants to embed sk8brd-proto doesn't have to duplicate the same ssh2 calls. When another protocol is added, I think a `sk8brd_connect()` func in sk8brd-proto that would accept a connection type parameter and return something that has `impl Write` would be in order. Then, all calls like `sess.set_blocking` would be to be abstracted as well, I suppose, but that's client specifics